packages/fast-usdc/src/exos/settler.js

```diff
@@ -1,97 +1,257 @@
+
+            default:
+              throw Error('TODO: think harder');
+          }
+        },
+      },
+      notify: {
+        /**
+         * @param {EvmHash} txHash
+         * @param {NobleAddress} sender
+         * @param {NatValue} amount
+         * @param {string} EUD
+         * @param {boolean} success
+         * @returns {void}
+         */
+        notifyAdvancingResult(txHash, sender, amount, EUD, success) {
+          const { mintedEarly } = this.state;
+          const key = makeMintedEarlyKey(sender, amount);
+          if (mintedEarly.has(key)) {
+            mintedEarly.delete(key);
+            if (success) {
+              void this.facets.self.disburse(txHash, sender, amount);
+            } else {
+              void this.facets.self.forward(txHash, sender, amount, EUD);
+            }
+          } else {
+            statusManager.advanceOutcome(sender, amount, success);
+          }
+        },
+      },
+      self: {
+        /**
+         * @param {EvmHash} txHash
+         * @param {NobleAddress} sender
+         * @param {NatValue} amount
+         */
+        async disburse(txHash, sender, amount) {
+          const { repayer, settlementAccount } = this.state;
+          const received = AmountMath.make(USDC, amount);
+          const { zcfSeat: settlingSeat } = zcf.makeEmptySeatKit();
+          const { calculateSplit } = makeFeeTools(feeConfig);
+          const split = calculateSplit(received);
+          trace('disbursing', split);
+
+          // TODO: what if this throws?
+          // arguably, it cannot. Even if deposits
+          // and notifications get out of order,
+          // we don't ever withdraw more than has been deposited.
+          await vowTools.when(
+            withdrawToSeat(
+              // @ts-expect-error Vow vs. Promise stuff. TODO: is this OK???
+              settlementAccount,
+              settlingSeat,
+              harden({ In: received }),
+            ),
+          );
+          zcf.atomicRearrange(
+            harden([[settlingSeat, settlingSeat, { In: received }, split]]),
           );
-        }
+          repayer.repay(settlingSeat, split);
+
+          // update status manager, marking tx `SETTLED`
+          statusManager.disbursed(txHash, sender, amount);
+        },
+        /**
+         * @param {EvmHash | undefined} txHash
+         * @param {NobleAddress} sender
+         * @param {NatValue} amount
+         * @param {string} EUD
+         */
+        async forward(txHash, sender, amount, EUD) {
+          const { settlementAccount } = this.state;
+
+          const dest = chainHub.makeChainAddress(EUD);
 
-        // TODO disperse funds
-        // ~1. fee to contractFeeAccount
-        // ~2. remainder in poolAccount
+          const txfrV = E(settlementAccount).transfer(
+            dest,
+            AmountMath.make(USDC, amount),
+          );
+          await vowTools.when(txfrV); // TODO: watch, handle failure
```

Comment:
This PR tackles a lot so I won't push to include here - but please create a follow up ticket or update this to refs that captures work needed for `Forwarding` and `ForwardFailed` states.

Reply:
We don't in general have issues for all TODOs, do we? Is this one special?